src/utils/types.ts

```diff
@@ -40,3 +40,23 @@ export type SuperiorPick<T, K extends keyof T> = T extends any ? {
  * Original Omit type extended to work on Union type objects
  */
 export type SuperiorOmit<T, K extends keyof any> = T extends any ? SuperiorPick<T, Exclude<keyof T, K>> : never;
+
+/**
+ * Extracts a transformed `ReadonlyMap` where the values are simplified to the `selfId` property
+ * of the objects if available. Otherwise returns the type as is.
+ *
+ * @template T - The type to transform, e.g. `ReadonlyMap<string, object>`.
+ * @returns A `ReadonlyMap` keyed similarly but with values as the `selfId` of the objects,
+ *          or the original type `T` if `T` is not a `ReadonlyMap` or the objects don't have `selfId`.
+ */
+export type ExtractSelfId<T> = T extends ReadonlyMap<string, infer U>
```

Comment:
this type is no longer used anywhere, let's delete it :)

Reply:
probably got introduced with the rebase, removed.